src/topic.ts

```diff
@@ -656,6 +664,78 @@ export class Topic {
     return this.publish(data, attributes, callback!);
   }
 
+  setMetadata(options: TopicMetadata, gaxOpts?: CallOptions):
+      Promise<SetTopicMetadataResponse>;
+  setMetadata(options: TopicMetadata, callback: SetTopicMetadataCallback): void;
+  setMetadata(
+      options: TopicMetadata, gaxOpts: CallOptions,
+      callback: SetTopicMetadataCallback): void;
+  /**
+   * @typedef {array} SetTopicMetadataResponse
+   * @property {object} 0 The full API response.
+   */
+  /**
+   * @callback SetTopicMetadataCallback
+   * @param {?Error} err Request error, if any.
+   * @param {object} apiResponse The full API response.
+   */
+  /**
+   * Updates the topic.
+   *
+   * @see [UpdateTopicRequest API Documentation]{@link https://cloud.google.com/pubsub/docs/reference/rest/v1/UpdateTopicRequest}
```

Comment:
from reading this documentation, am I reading correctly that labels are the only "meta information" you would potentially change on a topic, and this method since it's calling `updateTopic` would also potentially be able to change the `name` of a topic?

I wonder if a better method name would just be `updateTopic`? which would better match the upstream API too?

Reply:
`setMetadata` is the generic method name we use across all the libraries to update resource fields. I'm definitely not married to it or anything, just went with it for the sake of consistency. I'd be ok with moving away from some of our older conventions to better mirror the upstream apis.

@JustinBeckwith @stephenplusplus @jkwlui what do you guys think?